### PR TITLE
add new torchdata.nodes doc file

### PR DESCRIPTION
**Pull request by divyanshk**
Clone the repo on local, created the environment, updated the `/docs` folder, then`make html` and `sphinx-serve -b build`.

Screenshot 2024-12-09 at 3 46 59 PM

Screenshot 2024-12-09 at 3 47 24 PM

### CODE REVIEW

File: docs/source/torchdata.nodes.rst [hunk 1]
```diff
@@ -0,0 +1,10 @@
+:tocdepth: 3
+
+torchdata.nodes
+===================
+
+torchdata.nodes is a library of composable iterators (not iterables!) that let you chain together common dataloading and pre-proc operations. It follows a streaming programming model, although "sampler + Map-style" can still be configured if you desire.
+
+torchdata.nodes adds more flexibility to the standard torch.utils.data offering, and introduces multi-threaded parallelism in addition to multi-process (the only supported approach in torch.utils.data.DataLoader), as well as first-class support for mid-epoch checkpointing through a state_dict/load_state_dict interface.
+
+More information on torchdata.nodes can be found in the `README <https://github.com/pytorch/data/blob/main/torchdata/nodes/README.md>`_.
```
Comment (andrewkho) [inline docs/source/torchdata.nodes.rst:10]: Should we flip it and have README point here, and copy/paste most of the README here?

Comment (divyanshk) [reply docs/source/torchdata.nodes.rst:10]: yeah lets do that, let's keep it in both places for now ? 

i'll also edit the main readme to include a section on nodes

**Review by andrewkho — APPROVED** [review]
looks amazing!

**Comment by andrewkho** [comment]
Maybe for Tutorials/Examples we can have "Coming soon" or something for torchdata.nodes

**Comment by divyanshk** [comment]
@andrewkho Would it look a little off if we had a mostly empty page for "Tutorials and Examples > torchdata.nodes". If feel we can just add the page when we have the tutorial ready (which I believe is fairly soon). What say ?

I'll land this diff without it, but if you think we should add an empty page with "Coming soon!" do let me know, I'll create another PR. 

Also. I will add a note on tutorial coming soon in the overall README.md, so atleast that information is conveyed.

**Pull request merged**